third_party/terraform/tests/resource_compute_router_nat_test.go

```diff
@@ -0,0 +1,264 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccComputeRouterNat_basic(t *testing.T) {
+	t.Parallel()
+
+	testId := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterNatDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouterNatBasic(testId),
+				Check: testAccCheckComputeRouterNatExists(
+					"google_compute_router_nat.foobar"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_compute_router_nat.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			resource.TestStep{
+				Config: testAccComputeRouterNatKeepRouter(testId),
+				Check: testAccCheckComputeRouterNatDelete(
+					"google_compute_router_nat.foobar"),
+			},
+		},
+	})
+}
+
+func TestAccComputeRouterNat_withManualIpAndSubnetConfiguration(t *testing.T) {
+	t.Parallel()
+
+	testId := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterNatDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouterNatWithManualIpAndSubnetConfiguration(testId),
+				Check: testAccCheckComputeRouterNatExists(
+					"google_compute_router_nat.foobar"),
+			},
+		},
+	})
+}
+
+func testAccCheckComputeRouterNatDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	routersService := config.clientCompute.Routers
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_compute_router" {
```

Comment:
Is this supposed to be router or router nat?

Reply:
I think it can be either at this line, but you ultimately have to check that the Router has been deleted. I think the idea is that since the Nat is just a block within a Router, you need to check that the Router has been deleted at the end of the test.